sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -602,14 +602,25 @@ propertyList
     ;
 
 property
-    : key=propertyKey (EQ? value=propertyValue)?
+    : key=propertyKeyOrStringLit EQ value=propertyValue               #propertyWithKeyAndEquals
+    | key=propertyKeyNoCoalesce value=propertyValue?                  #propertyWithStringKeyNoEquals
+    | key=propertyKey (EQ? value=propertyValue)?                      #propertyWithIdentifierKey
```

Comment:
hmm do we still need this rule? I think the above two covers all the supported cases?